eclipse/sw360
# feat(CRUI):Change CR state 'On Hold' to 'Awaiting Response' & edit PreferredClearingDate

[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.

- Changed `Clearing Request` state from `On Hold` to `Awaiting Response`.
- Requesting user can now change the `preferred clearing date` (RequestedClearingDate) to future date in `Edit` mode.

Issue: #1560 


### How To Test?

1. Run the migration script in the PR, and all the Clearing Request with `On Hold` status should get changed to `Awaiting Response`
2. Login as a normal user & create a `Clearing Request` for a `Project` from Project portlet homepage or `License Clearing` tab of project details page.
3. Edit the Clearing Request created in step 2, and modify the `Preferred Clearing Date`.
4. Only `Requesting user` should be able to change the `Preferred Clearing Date` to future date only.
5. Login with `Admin` or `Clearing Expert` user role.
6. Edit the `Clearing Request` and check that Clearing Request State dropdown field has `Awaiting Response` option instead of `On Hold`.
7. All the tooltips should also show the `Awaiting Response` option instead of `On Hold`.
8. Make sure that the project, and all the user used in testing the PR belongs to the same `Group` (department).

### Checklist
Must:

Signed-off-by: Abdul Kapti <abdul.kapti@siemens-healthineers.com>
